x-pack/legacy/plugins/code/public/utils/saga_utils.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { take, spawn } from 'redux-saga/effects';
+import { Action } from 'redux-actions';
+
+function* cancelRequest(action: Action<any>, abortController: AbortController) {
+  yield take(action.type);
+  abortController.abort();
+}
+
+export const cancellableSaga = (saga: any) =>
+  function*(action: Action<any>) {
+    const abortController = new AbortController();
```

Comment:
Correct me if I'm wrong. Is this `abortController` created every time? Shouldn't we cancel the one last created when the request is sent?

Reply:
My understanding here is the next action with the same type will trigger the `abort()` (line 12) of a previous created `abortController`. Then thus signal via line 19. So essentially this code block is just hooking up all the cancellation logic, which the actual cancellation is triggered until the next action with the same type is detected. It does take a moment to understand the code here. @WangQianliang if you reference any stackoverflow answer, please comment here.

Reply:
Yes, it's created every time to detect if there's new action dispatched so that it will cancel this request. In my latest commit, it would cancel the abort request task if api request succeeds or fails.